frontend/ui/widget/textviewer.lua

```diff
@@ -531,6 +540,90 @@ function TextViewer:handleTextSelection(text, hold_duration, start_idx, end_idx,
     end
 end
 
+function TextViewer:reinit()
+    local low, high = self.scroll_text_w.text_widget:getVisibleHeightRatios() -- try to keep position
+    local ratio = low == 0 and 0 or (low + high) / 2 -- if we are at the beginning, keep the first line visible
+    self:init(true) -- do not add default buttons once more
+    self:onShow()
+    self.scroll_text_w:scrollToRatio(ratio, ratio == 0)
+end
+
+function TextViewer.saveSetting(setting, value)
+    G_reader_settings:saveSetting(setting, value)
+    UIManager:show(Notification:new{
+        text = _("Text viewer setting saved as default."),
+    })
+end
+
+function TextViewer:showMenu()
+    local dialog
+    local buttons = {
+        {{
+            text_func = function()
+                return T(_("Font size: %1"), self.text_font_size)
+            end,
+            align = "left",
+            callback = function()
+                UIManager:close(dialog)
+                local SpinWidget = require("ui/widget/spinwidget")
+                local widget = SpinWidget:new{
+                    title_text = _("Font size"),
+                    value = self.text_font_size,
+                    value_min = 12,
+                    value_max = 30,
+                    default_value = 20, -- x_smallinfofont
+                    keep_shown_on_apply = true,
+                    callback = function(spin)
+                        self.text_font_size = spin.value
+                        self:reinit()
+                    end,
+                }
+                UIManager:show(widget)
+            end,
+            hold_callback = function()
+                TextViewer.saveSetting("textviewer_font_size", self.text_font_size)
+            end,
+        }},
+        {{
+            text_func = function()
+                return _("Monospace font") .. (self.monospace_font and "  \u{2713}" or "  \u{2003}")
+            end,
```

Comment:
Dunno if you witnessed some odd stuff while testing, but sure you don't need the ugly trick I had to use with this checkbox:
https://github.com/koreader/koreader/blob/13a521c39849459cfbbd69e01f17c0c601d7dfbd/frontend/ui/widget/bookmapwidget.lua#L1204-L1229

Or is it because all such checkboxes calls :reinit(), which will rebuilt this popup menu ?
(If yes, I guess everything is rebuild and redrawn, and that the menu popup is closed as soon as you toggle/change things ? Which is a bit different than the behaviour in PageBrowser/Bookmap where we just update the content, and the popup menu stays open. But ok if it's too complicated to get this same behaviour?)

Reply:
Yes, the menu is closed on tap.
If we tap Font size, we see the SpinWidget (that remains open on apply) and we need to see more text to choose the size.
If we tap Monospace or Justify, we see the immediate effect on text, and what is our most frequent next action?
If it is reading, that's okay to close the menu.
If it is switching the setting back, it's better to keep it open.
If it is long-pressing to set the setting as default, it's better to keep it open.